easybuild/tools/filetools.py

```diff
@@ -298,9 +298,8 @@ def symlink(source_path, symlink_path, use_abspath_source=True):
     if use_abspath_source:
         source_path = os.path.abspath(source_path)
 
-    if os.path.exists(symlink_path):
+    if os.path.exists(symlink_path) and os.path.abspath(source_path) == os.path.abspath(os.readlink(symlink_path)):
```

Comment:
You should rather bail out if they aren't the same otherwise the else below will be doing the wrong thing.